tutorial/westeros/westeros.ipynb

```diff
@@ -0,0 +1,1268 @@
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Working backwards along the Reference Energy System, we can add connections for the `bulb`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "bulb_out = make_df(base_output, technology='bulb', commodity='light', level='useful', value=1.0)\n",
+    "scenario.add_par('output', bulb_out)\n",
+    "\n",
+    "bulb_in = make_df(base_input, technology='bulb', commodity='electricity',  level='final', value=1.0, unit='%')\n",
+    "scenario.add_par('input', bulb_in)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "Next, the `grid`, with loses of 13%"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "grid_efficiency = 0.87\n",
+    "grid_out = make_df(base_output, technology='grid', commodity='electricity', level='final', value=grid_efficiency)\n",
+    "scenario.add_par('output', grid_out)\n",
+    "\n",
+    "grid_in = make_df(base_input, technology='grid', commodity='electricity',  level='secondary', value=1.0, unit='%')\n",
+    "scenario.add_par('input', grid_in)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "And finally, our power plants which are not resource constrained in this simple model thus have no `input`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "coal_out = make_df(base_output, technology='coal_ppl', commodity='electricity', level='secondary', value=1.)\n",
+    "scenario.add_par('output', coal_out)\n",
+    "\n",
+    "wind_out = make_df(base_output, technology='wind_ppl', commodity='electricity', level='secondary', value=1.)\n",
+    "scenario.add_par('output', wind_out)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Operational Constraints and Parameters"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "The model has a number of \"reality\" constraints, which relate built capacity to available power.\n",
+    "\n",
+    "The `CAPACITY_CONSTRAINT` \n",
+    "\n",
+    "$\\sum_{m} ACT_{n,t,y^V,y,m,h}\n",
+    "    \\leq duration\\_time_{h} \\cdot capacity\\_factor_{n,t,y^V,y,h} \\cdot CAP_{n,t,y^V,y}\n",
+    "    \\quad t \\ \\in \\ T^{INV}$\n",
+    "    \n",
+    "TODO: needed? asked daniel, wait for reply\n",
```

Comment:
`OPERATION_CONSTRAINT` is only needed if you use sub-annual timeslices, to distinguish between required down-time per `time` step compared to flexible down-time throughout the year - e.g., when you can decide whether to do maintenance for a plant in summer or winter.

Reply:
done!